ChangeLog

```diff
@@ -3,12 +3,106 @@ Changelog
 
 ## v2.4.0
 
-- Enum: uniq is modified to use structural comparison internally, instead
-    of previously used physical comparison, which was most likely a bug.
-    The old behavior is provided by the new Enum.uniqq function.
-    Enum.uniq_by was added for additional convenience.
+- ignore deprecation warnings
+  (Gabriel Scherer)
+- BatBitSet: use Bytes instead of String
+  (Gabriel Scherer)
+- rm qtest-generated file all_tests.ml upon 'make clean'
+  (Francois Berenger)
+- corrected many warnings
+  (Francois Berenger)
+- regression tests for issue #609
+  (Gabriel Scherer)
+- Makefile: create testsuite-only-{byte,native} targets
+  (Gabriel Scherer)
+- BatHashtbl: fix hash computation bug introduced by 4.01
+  (Gabriel Scherer)
+- BatText: synchronize nsplit to match BatString.nsplit
+  (Gabriel Scherer)
+- BatInt: repared quick check test for mid
+  (Francois Berenger)
+- updated NEXT_RELEASE tags
+  (Francois Berenger)
+- BatLazyList: fix remove_all_such documentation
+  (Xavier Van de Woestyne)
+- BatMap: better ocamldoc for update
+  (Francois Berenger)
+- BatMap: add pop_min_binding and pop_max_binding
+  (Francois Berenger)
+- BatMap: fix at_rank_exn compilation
+  (Gabriel Scherer)
+- BatMap: modified update to behave as Set.update
+  (Francois Berenger)
+- BatMap: add at_rank_exn
+  (Francois Berenger)
+- BatSet: shorter exception strings in at_rank_exn
+  (Francois Berenger)
+- BatMap: add update
+  (Francois Berenger)
+- BatEnum: add interleave utility function
+  (Aleksey Z. Arens)
+- fix _tags warnings settings
+  (Gabriel Scherer)
+- BatFormat: expose asprintf for V>=4.01
+  (Ralf Vogler)
+- BatSet: more generic version of update
```

Comment:
what does this mean? What was the previous version, why is the new version more generic?